// unstyled/dist/unstyled.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var styled = _interopDefault(require('styled-components'));
var propTypes = require('prop-types');
var React = _interopDefault(require('react'));

/* Basic theme with minimal styling. Nearly unstyled.*/
require('typeface-lato');

var DefaultTheme = {
  color: {
    primary: 'rebeccapurple',
    secondary: '#587291',
    success: '#59B655',
    alert: '#CA3B33',
    warning: '#F3BE43',
    info: '#3E85CA',
    grey: '#E8E8E8'
  },
  typography: {
    fontColor: 'rgba(0, 0, 0, 0.84)',
    fontFamily: 'Lato,Open Sans,Helvetica,Arial,sans-serif',
    fontSize: 'inherited',
    fontWeight: 400,
    lineHeight: '1rem'
  },
  heading: {
    fontSize: '1.6rem',
    fontWeight: 600
  }
};

/* Theme based on the `styed-components` website.*/
var Theme = {
  primaryColor: "#CE7692",
  secondaryColor: "#FDEFD8",
  fontSize: "16px"
};

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _taggedTemplateLiteral(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  return Object.freeze(Object.defineProperties(strings, {
    raw: {
      value: Object.freeze(raw)
    }
  }));
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: -webkit-box;\n  display: -ms-flexbox;\n  display: flex;\n  -webkit-box-orient: horizontal;\n  -webkit-box-direction: normal;\n  -ms-flex-direction: row;\n  flex-direction: row;\n  -ms-flex-wrap: wrap;\n  flex-wrap: wrap;\n  -webkit-box-align: stretch;\n  -ms-flex-align: stretch;\n  align-items: stretch;\n  padding: 0em;\n\n  // Calculate width for contained columns.\n  ", " {\n    width: ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: inline-block;\n  font-family: ", ";\n  padding-left: 1rem;\n  padding-right: 1rem;\n  position: relative;\n  vertical-align: top;\n  width: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * A Column of the Grid.
 */

var Column = styled.div(_templateObject(), function (props) {
  return props.theme.typography.fontFamily;
}, function (_ref) {
  var width = _ref.width;
  return width;
});
/**
 * Container component for the Grid.
 */

var Grid = styled.div(_templateObject2(), Column, function (_ref2) {
  var columns = _ref2.columns;
  return columns ? 100 / columns + '%' : '6.25%';
});
Column.defaultProps = {
  theme: DefaultTheme
};
Grid.Column = Column;

function _templateObject2$1() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  border: ", " solid\n    ", ";\n  border-radius: ", ";\n  flex-direction: ", ";\n  margin: 1rem 0;\n  min-height: 2.875em;\n  width: ", ";\n  //-webkit-box-shadow: 0px 0px 0px 1px rgba(34, 36, 38, 0.22) inset,\n  //  0px 0px 0px 0px rgba(0, 0, 0, 0);\n  //box-shadow: 0px 0px 0px 1px rgba(34, 36, 38, 0.22) inset,\n  //  0px 0px 0px 0px rgba(0, 0, 0, 0);\n  ", " {\n    border-width: ", ";\n    color: ", ";\n    display: ", ";\n  }\n\n  &:first-child {\n    margin-top: 0rem;\n  }\n  &:last-child {\n    margin-bottom: 0rem;\n  }\n"]);

  _templateObject2$1 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$1() {
  var data = _taggedTemplateLiteral(["\n  align-items: center;\n  border-right: 1px solid rgba(34, 36, 38, 0.22);\n  color: ", ";\n  cursor: pointer;\n  display: -webkit-box;\n  display: -ms-flexbox;\n  display: flex;\n  font-family: ", ";\n  font-weight: ", ";\n  line-height: 1;\n  padding: 0.9375em 1.125em;\n  position: relative;\n  text-decoration: none;\n  vertical-align: middle;\n  -webkit-box-flex: 0;\n  -webkit-box-align: center;\n  -ms-flex: 0 0 auto;\n  -ms-flex-align: center;\n  &:hover {\n    background-color: rgba(0, 0, 0, 0.05);\n    color: rgba(0, 0, 0, 0.95);\n  }\n  &:active {\n    color: ", ";\n  }\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}
var MenuItem = styled.a(_templateObject$1(), function (_ref) {
  var theme = _ref.theme;
  return theme.black;
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return props.bold ? '600' : '400';
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.black;
});
var Menu = styled.div(_templateObject2$1(), function (props) {
  return props.borderless ? 0 : '1px';
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.grey;
}, function (_ref4) {
  var rounded = _ref4.rounded;
  return rounded ? '0.15rem' : 0;
}, function (props) {
  return props.vertical ? 'column' : 'row';
}, function (props) {
  return props.vertical ? '15rem' : 'auto';
}, MenuItem, function (props) {
  return props.borderless ? 0 : '1px';
}, function (props) {
  return props.color || 'black';
}, function (props) {
  return props.vertical ? 'block' : 'inline';
}); // Props

Menu.propTypes = {
  rounded: propTypes.PropTypes.bool
};
MenuItem.propTypes = {
  bold: propTypes.PropTypes.bool
};
Menu.defaultProps = {
  theme: DefaultTheme
};
MenuItem.defaultProps = {
  theme: DefaultTheme
};
Menu.Item = MenuItem;

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  opacity: 0.85;\n  margin: 0.75em 0em;\n  &:first-child {\n    margin-top: 0em;\n  }\n  &:last-child {\n    margin-bottom: 0em;\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2$2() {
  var data = _taggedTemplateLiteral(["\n  display: block;\n  font-family: 'Lato', 'Helvetica Neue', Arial, Helvetica, sans-serif;\n  font-weight: bold;\n  margin: -0.14285714em 0em 0rem 0em;\n"]);

  _templateObject2$2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$2() {
  var data = _taggedTemplateLiteral(["\n  background: ", ";\n  border-radius: ", ";\n  color: ", ";\n  font-family: ", ";\n  line-height: 1.4285em;\n  min-height: 1em;\n  margin: 1em 0em;\n  padding: 1em 1.5em;\n  position: relative;\n  //color: rgba(0, 0, 0, 0.87);\n  -webkit-transition: opacity 0.1s ease, color 0.1s ease, background 0.1s ease,\n    -webkit-box-shadow 0.1s ease;\n  transition: opacity 0.1s ease, color 0.1s ease, background 0.1s ease,\n    -webkit-box-shadow 0.1s ease;\n  transition: opacity 0.1s ease, color 0.1s ease, background 0.1s ease,\n    box-shadow 0.1s ease;\n  transition: opacity 0.1s ease, color 0.1s ease, background 0.1s ease,\n    box-shadow 0.1s ease, -webkit-box-shadow 0.1s ease;\n  -webkit-box-shadow: 0px 0px 0px 1px rgba(34, 36, 38, 0.22) inset,\n    0px 0px 0px 0px\n      ", ";\n  box-shadow: 0px 0px 0px 1px\n      ", " inset,\n    0px 0px 0px 0px\n      ", ";\n  &:first-child {\n    margin-top: 0em;\n  }\n  &:last-child {\n    margin-bottom: 0em;\n  }\n"]);

  _templateObject$2 = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * The Message container.
 */

var Message = styled.div(_templateObject$2(), function (_ref) {
  var theme = _ref.theme;
  return theme.secondaryColor;
}, function (_ref2) {
  var rounded = _ref2.rounded;
  return rounded ? '0.15rem' : 0;
}, function (_ref3) {
  var theme = _ref3.theme,
      color = _ref3.color;
  return color ? theme[color] : theme.primaryColor;
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (_ref4) {
  var theme = _ref4.theme,
      color = _ref4.color;
  return color ? theme[color] : theme.primaryColor;
}, function (_ref5) {
  var theme = _ref5.theme,
      color = _ref5.color;
  return color ? theme[color] : theme.primaryColor;
}, function (_ref6) {
  var theme = _ref6.theme,
      color = _ref6.color;
  return color ? theme[color] : theme.primaryColor;
});
var Header = styled.div(_templateObject2$2());
var Text = styled.p(_templateObject3()); // Props

Message.propTypes = {
  color: propTypes.PropTypes.string,
  rounded: propTypes.PropTypes.bool // Default Theme

};
Message.defaultProps = {
  theme: DefaultTheme // Assign children

};
Message.Header = Header;
Message.Text = Text;

function _templateObject$3() {
  var data = _taggedTemplateLiteral(["\n  background-color: ", ";\n  border: none;\n  border-radius: 0.125em;\n  color: ", ";\n  display: inline-block;\n  font-family: ", ";\n  font-size: 1em;\n  font-style: normal;\n  font-weight: bold;\n  line-height: 1em;\n  margin: 0em 0.125em;\n  min-height: 1em;\n  padding: 0.5833em 0.833em;\n  text-align: center;\n  text-decoration: none;\n  vertical-align: baseline;\n"]);

  _templateObject$3 = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * A Badge displays a highlighted number.
 * TODO add chroma.js lib to get inverted colors
 */

var Badge = styled.span(_templateObject$3(), function (_ref) {
  var theme = _ref.theme;
  return theme.secondaryColor;
}, function (props) {
  return props.color ? props.color : props.theme.typography.primaryColor;
}, function (props) {
  return props.theme.typography.fontFamily;
}); // Default Theme

Badge.defaultProps = {
  theme: DefaultTheme
};

// const colors = require(`./colors`).default

function _templateObject$4() {
  var data = _taggedTemplateLiteral(["\n  display: block;\n  font-family: ", ";\n  margin: 0 auto;\n  max-width: 960px;\n\n  /* Mobile */\n  @media only screen and (max-width: 767px) {\n    width: auto;\n    margin-left: 1em;\n    margin-right: 1em;\n  }\n\n  /* Tablet */\n  @media only screen and (min-width: 768px) and (max-width: 991px) {\n    width: 723px;\n    margin-left: auto !important;\n    margin-right: auto !important;\n  }\n\n  /* Small Monitor */\n  @media only screen and (min-width: 992px) and (max-width: 1199px) {\n    width: 933px;\n    margin-left: auto !important;\n    margin-right: auto !important;\n  }\n\n  /* Large Monitor */\n  @media only screen and (min-width: 1200px) {\n    width: 1127px;\n    margin-left: auto !important;\n    margin-right: auto !important;\n  }\n"]);

  _templateObject$4 = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * A Container limits it's content to a maximum width.
 */

var Container = styled.div(_templateObject$4(), function (props) {
  return props.theme.typography.fontFamily;
});
Container.defaultProps = {
  theme: DefaultTheme
};

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  display: inline-block;\n  font-family: ", ";\n  position: relative;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3$1() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n  cursor: pointer;\n  display: inline-block;\n  font-family: ", ";\n  opacity: 1;\n  outline: 0;\n  visibility: visible;\n  :focus {\n    /* clicking on label should toggle the menu */\n    pointer-events: none;\n    border: none;\n  }\n  :focus ~ ", " {\n    opacity: 1;\n    visibility: visible;\n    pointer-events: auto;\n  }\n"]);

  _templateObject3$1 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2$3() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n  display: inline-block;\n  font-family: ", ";\n  padding: 0.5rem;\n  white-space: nowrap;\n"]);

  _templateObject2$3 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$5() {
  var data = _taggedTemplateLiteral(["\n  background-color: white;\n  border: 1px solid ", ";\n  border-radius: ", ";\n  display: block;\n  margin-top: 0.5rem;\n  opacity: 0;\n  position: absolute;\n  transition: visibility 0.5s;\n  visibility: hidden;\n  z-index: 1;\n"]);

  _templateObject$5 = function _templateObject() {
    return data;
  };

  return data;
}
var DropdownMenu = styled.div(_templateObject$5(), function (props) {
  return props.theme.color.grey;
}, function (props) {
  return props.rounded ? '0.15rem' : null;
});
var DropdownItem = styled.div(_templateObject2$3(), function (props) {
  return props.color || props.theme.typography.fontColor;
}, function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
});
var DropdownHeader = styled.div.attrs({
  tabIndex: 0
})(_templateObject3$1(), function (props) {
  return props.color || props.theme.typography.fontColor;
}, function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
}, DropdownMenu);
var Dropdown = styled.div(_templateObject4(), function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
});
Dropdown.defaultProps = {
  theme: DefaultTheme
};
DropdownMenu.defaultProps = {
  theme: DefaultTheme
};
DropdownHeader.defaultProps = {
  theme: DefaultTheme
};
DropdownItem.defaultProps = {
  theme: DefaultTheme
};
Dropdown.Menu = DropdownMenu;
Dropdown.Header = DropdownHeader;
Dropdown.Item = DropdownItem;

function _templateObject$6() {
  var data = _taggedTemplateLiteral(["\n  border: none;\n  color: ", ";\n  font-family: ", ";\n  font-weight: bold;\n  line-height: 1.28571429em;\n  margin: calc(2rem - 0.14285714em) 0em 1rem;\n  padding: 0em 0em;\n  text-transform: none;\n"]);

  _templateObject$6 = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * A major title block.
 */

var Header$1 = styled.div(_templateObject$6(), function (props) {
  return props.color || props.theme.typography.fontColor;
}, function (props) {
  return props.theme.typography.fontFamily;
});
Header$1.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$7() {
  var data = _taggedTemplateLiteral(["\n  background-color: ", ";\n  border: none;\n  border-radius: 0.125em;\n  // Use white font color if contrast between background and standard color i not below 7\n  color: ", ";\n  // chroma.contrast(\n  //   props.theme.color[props.color] || 'white',\n  //   props.theme.typography.fontColor\n  // ) > 7\n  //   ? props.theme.typography.fontColor\n  //   : 'white'};\n  display: inline-block;\n  font-family: ", ";\n  font-size: 1em;\n  font-style: normal;\n  font-weight: bold;\n  line-height: 1em;\n  margin: 0em 0.125em;\n  min-height: 1em;\n  padding: 0.5833em 0.833em;\n  text-align: center;\n  text-decoration: none;\n  vertical-align: baseline;\n"]);

  _templateObject$7 = function _templateObject() {
    return data;
  };

  return data;
}

var chroma = require('chroma-js');
/**
 * A Basic Label representing a HTML form label.
 * TODO refactor Label as Form.Label and use this as general label component
 */


var Label = styled.span(_templateObject$7(), function (props) {
  return props.color ? props.theme.color[props.color] : props.theme.color.grey;
}, function (props) {
  return props.color && chroma(props.theme.color[props.color]).luminance() < 0.5 ? 'white' : props.theme.typography.fontColor;
}, function (props) {
  return props.theme.typography.fontFamily;
}); // Default Theme

Label.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$8() {
  var data = _taggedTemplateLiteral(["\n  display: block;\n  left: 50%;\n  margin: 0px;\n  position: absolute;\n  text-align: center;\n  top: 50%;\n  z-index: 1000;\n  /* 'transform' (%) doesn't seem to work with 'animation'. \n  Fixing it for now with an absolute value.\n  TODO: Solve with relative value (%).*/\n  -webkit-transform: translateY(-1.1rem);\n  transform: translateY(-1.1rem);\n\n  /* Static Shape */\n  &:before {\n    border-radius: 500rem;\n    border: 0.2em solid rgba(0, 0, 0, 0.1);\n    content: '';\n    height: 100%;\n    left: 50%;\n    position: absolute;\n    top: 0%;\n    width: 100%;\n  }\n\n  /* Active Shape */\n  &:after {\n    content: '';\n    height: 100%;\n    left: 50%;\n    border-radius: 500rem;\n    border-color: #767676 transparent transparent;\n    border-style: solid;\n    border-width: 0.2em;\n    box-shadow: 0px 0px 0px 1px transparent;\n    position: absolute;\n    top: 0%;\n    width: 100%;\n    animation: loader 0.6s linear;\n    animation-iteration-count: infinite;\n    -webkit-animation: loader 0.6s linear;\n    -webkit-animation-iteration-count: infinite;\n  }\n\n  /* Active Animation */\n  @-webkit-keyframes loader {\n    from {\n      -webkit-transform: rotate(0deg);\n      transform: rotate(0deg);\n    }\n    to {\n      -webkit-transform: rotate(360deg);\n      transform: rotate(360deg);\n    }\n  }\n  @keyframes loader {\n    from {\n      -webkit-transform: rotate(0deg);\n      transform: rotate(0deg);\n    }\n    to {\n      -webkit-transform: rotate(360deg);\n      transform: rotate(360deg);\n    }\n  }\n\n  &:before,\n  &:after {\n    height: 2.2rem;\n    margin: 0em 0em 0em -1.14rem;\n    width: 2.2rem;\n  }\n\n  /*!* Sizes *!\n.ui.mini.loader:before,\n.ui.mini.loader:after {\n  width: 1rem;\n  height: 1rem;\n  margin: 0em 0em 0em -0.5rem;\n}\n.ui.tiny.loader:before,\n.ui.tiny.loader:after {\n  width: 1.14285714rem;\n  height: 1.14285714rem;\n  margin: 0em 0em 0em -0.57142857rem;\n}\n.ui.small.loader:before,\n.ui.small.loader:after {\n  width: 1.71428571rem;\n  height: 1.71428571rem;\n  margin: 0em 0em 0em -0.85714286rem;\n}\n.ui.loader:before,\n.ui.loader:after {\n  width: 2.28571429rem;\n  height: 2.28571429rem;\n  margin: 0em 0em 0em -1.14285714rem;\n}\n.ui.large.loader:before,\n.ui.large.loader:after {\n  width: 3.42857143rem;\n  height: 3.42857143rem;\n  margin: 0em 0em 0em -1.71428571rem;\n}\n.ui.big.loader:before,\n.ui.big.loader:after {\n  width: 3.71428571rem;\n  height: 3.71428571rem;\n  margin: 0em 0em 0em -1.85714286rem;\n}\n.ui.huge.loader:before,\n.ui.huge.loader:after {\n  width: 4.14285714rem;\n  height: 4.14285714rem;\n  margin: 0em 0em 0em -2.07142857rem;\n}\n.ui.massive.loader:before,\n.ui.massive.loader:after {\n  width: 4.57142857rem;\n  height: 4.57142857rem;\n  margin: 0em 0em 0em -2.28571429rem;\n}\n\n!*-------------------\n      Coupling\n--------------------*!\n\n\n!* Show inside active dimmer *!\n.ui.dimmer .loader {\n  display: block;\n}\n\n!* Black Dimmer *!\n.ui.dimmer .ui.loader {\n  color: rgba(255, 255, 255, 0.9);\n}\n.ui.dimmer .ui.loader:before {\n  border-color: rgba(255, 255, 255, 0.15);\n}\n.ui.dimmer .ui.loader:after {\n  border-color: #FFFFFF transparent transparent;\n}\n\n!* White Dimmer (Inverted) *!\n.ui.inverted.dimmer .ui.loader {\n  color: rgba(0, 0, 0, 0.87);\n}\n.ui.inverted.dimmer .ui.loader:before {\n  border-color: rgba(0, 0, 0, 0.1);\n}\n.ui.inverted.dimmer .ui.loader:after {\n  border-color: #767676 transparent transparent;\n}\n\n\n!*******************************\n             Types\n*******************************!\n\n\n!*-------------------\n        Text\n--------------------*!\n\n.ui.text.loader {\n  width: auto !important;\n  height: auto !important;\n  text-align: center;\n  font-style: normal;\n}\n\n\n!*******************************\n            States\n*******************************!\n\n.ui.indeterminate.loader:after {\n  -webkit-animation-direction: reverse;\n          animation-direction: reverse;\n  -webkit-animation-duration: 1.2s;\n          animation-duration: 1.2s;\n}\n.ui.loader.active,\n.ui.loader.visible {\n  display: block;\n}\n.ui.loader.disabled,\n.ui.loader.hidden {\n  display: none;\n}\n\n\n!*******************************\n            Variations\n*******************************!\n\n\n!*-------------------\n        Sizes\n--------------------*!\n\n\n!* Loader *!\n.ui.inverted.dimmer .ui.mini.loader,\n.ui.mini.loader {\n  width: 1rem;\n  height: 1rem;\n  font-size: 0.78571429em;\n}\n.ui.inverted.dimmer .ui.tiny.loader,\n.ui.tiny.loader {\n  width: 1.14285714rem;\n  height: 1.14285714rem;\n  font-size: 0.85714286em;\n}\n.ui.inverted.dimmer .ui.small.loader,\n.ui.small.loader {\n  width: 1.71428571rem;\n  height: 1.71428571rem;\n  font-size: 0.92857143em;\n}\n.ui.inverted.dimmer .ui.loader,\n.ui.loader {\n  width: 2.28571429rem;\n  height: 2.28571429rem;\n  font-size: 1em;\n}\n.ui.inverted.dimmer .ui.large.loader,\n.ui.large.loader {\n  width: 3.42857143rem;\n  height: 3.42857143rem;\n  font-size: 1.14285714em;\n}\n.ui.inverted.dimmer .ui.big.loader,\n.ui.big.loader {\n  width: 3.71428571rem;\n  height: 3.71428571rem;\n  font-size: 1.28571429em;\n}\n.ui.inverted.dimmer .ui.huge.loader,\n.ui.huge.loader {\n  width: 4.14285714rem;\n  height: 4.14285714rem;\n  font-size: 1.42857143em;\n}\n.ui.inverted.dimmer .ui.massive.loader,\n.ui.massive.loader {\n  width: 4.57142857rem;\n  height: 4.57142857rem;\n  font-size: 1.71428571em;\n}\n\n!* Text Loader *!\n.ui.mini.text.loader {\n  min-width: 1rem;\n  padding-top: 1.78571429rem;\n}\n.ui.tiny.text.loader {\n  min-width: 1.14285714rem;\n  padding-top: 1.92857143rem;\n}\n.ui.small.text.loader {\n  min-width: 1.71428571rem;\n  padding-top: 2.5rem;\n}\n.ui.text.loader {\n  min-width: 2.28571429rem;\n  padding-top: 3.07142857rem;\n}\n.ui.large.text.loader {\n  min-width: 3.42857143rem;\n  padding-top: 4.21428571rem;\n}\n.ui.big.text.loader {\n  min-width: 3.71428571rem;\n  padding-top: 4.5rem;\n}\n.ui.huge.text.loader {\n  min-width: 4.14285714rem;\n  padding*/\n"]);

  _templateObject$8 = function _templateObject() {
    return data;
  };

  return data;
}
var Loader = styled.div(_templateObject$8());

function _templateObject$9() {
  var data = _taggedTemplateLiteral(["\n  border-bottom: 1px solid rgba(34, 36, 38, 0.15);\n  font-family: ", ";\n  line-height: 2.875em;\n  min-height: 2.875em;\n  padding: 0.5rem 1rem;\n  position: relative;\n"]);

  _templateObject$9 = function _templateObject() {
    return data;
  };

  return data;
}
var NavBar = styled.div(_templateObject$9(), function (props) {
  return props.theme.typography.fontFamily;
});
NavBar.defaultProps = {
  theme: DefaultTheme
};

function _templateObject2$4() {
  var data = _taggedTemplateLiteral(["\n  border: 1px solid ", ";\n  box-sizing: border-box;\n  display: block;\n  margin-left: -1px;\n  padding: 0.5rem 0.75rem;\n  position: relative;\n  &:first-child {\n    margin-left: 0;\n    border-top-left-radius: 0.25rem;\n    border-bottom-left-radius: 0.25rem;\n  }\n  &:last-child {\n    border-top-right-radius: 0.25rem;\n    border-bottom-right-radius: 0.25rem;\n  }\n"]);

  _templateObject2$4 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$a() {
  var data = _taggedTemplateLiteral(["\n  border-radius: 0.25rem;\n  box-sizing: border-box;\n  display: flex;\n  flex-direction: row;\n  font-family: ", ";\n  list-style: none;\n  margin: 0;\n  padding-left: 0;\n  text-decoration: none;\n  width: auto;\n"]);

  _templateObject$a = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Pagination component.
 */

var Pagination = styled.ul(_templateObject$a(), function (props) {
  return props.theme.typography.fontFamily;
});
/**
 * Item inside Pagination.
 */

var PaginationItem = styled.li(_templateObject2$4(), function (props) {
  return props.theme.blue;
});
Pagination.defaultProps = {
  theme: DefaultTheme
};
PaginationItem.defaultProps = {
  theme: DefaultTheme
};
Pagination.Item = PaginationItem;

function _templateObject$b() {
  var data = _taggedTemplateLiteral(["\n  background: #ffffff;\n  border-radius: ", ";\n  border: 1px solid rgba(34, 36, 38, 0.15);\n  box-shadow: 0px 1px 2px 0 rgba(34, 36, 38, 0.15);\n  font-family: ", ";\n  margin: 1rem 0em;\n  padding: 1em 1em;\n  position: relative;\n  -webkit-box-shadow: 0px 1px 2px 0 rgba(34, 36, 38, 0.15);\n"]);

  _templateObject$b = function _templateObject() {
    return data;
  };

  return data;
}
var Segment = styled.div(_templateObject$b(), function (props) {
  return props.rounded ? '0.25rem' : 0;
}, function (props) {
  return props.theme.typography.fontFamily;
});
Segment.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$c() {
  var data = _taggedTemplateLiteral(["\n  background-color: ", ";\n  border-radius: 0;\n  color: ", ";\n  height: 100%;\n  left: 0;\n  top: 0;\n  max-height: 100%;\n  margin: 0;\n  overflow-y: auto;\n  position: fixed;\n  width: 250px;\n  z-index: 100;\n"]);

  _templateObject$c = function _templateObject() {
    return data;
  };

  return data;
}
var Sidebar = styled.div(_templateObject$c(), function (props) {
  return props.inverted ? props.theme.color.grey : 'white';
}, function (props) {
  return props.inverted ? 'white' : props.theme.color.grey;
}); // Props

Sidebar.propTypes = {
  rounded: propTypes.PropTypes.bool // Default Theme

};
Sidebar.defaultProps = {
  theme: DefaultTheme
};

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  display: block;\n  font-family: ", ";\n  margin-top: 0.5em;\n  width: inherit;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4$1() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  color: rgba(0, 0, 0, 0.4);\n  margin-top: 0.5em;\n"]);

  _templateObject4$1 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3$2() {
  var data = _taggedTemplateLiteral(["\n  color: rgba(0, 0, 0, 0.85);\n  display: block;\n  font-family: ", ";\n  font-weight: bold;\n  line-height: 1em;\n  font-size: 1.3em;\n  width: inherit;\n"]);

  _templateObject3$2 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2$5() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  padding: 0.5833em 0.833em;\n"]);

  _templateObject2$5 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$d() {
  var data = _taggedTemplateLiteral(["\n  border: 1px solid #e8e8e8;\n  border-radius: ", ";\n  color: rgba(0, 0, 0, 0.6);\n  display: flex;\n  font-family: ", ";\n  flex-direction: column;\n  flex-wrap: wrap;\n  font-size: 1em;\n  font-style: normal;\n  line-height: 1em;\n  margin: 0em 0.125em;\n  max-width: 100%;\n  position: relative;\n  vertical-align: baseline;\n  width: calc(960px / 3);\n"]);

  _templateObject$d = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * The wrapping Card component.
 */

var Card = styled.div(_templateObject$d(), function (_ref) {
  var rounded = _ref.rounded;
  return rounded ? '0.25rem' : 0;
}, function (props) {
  return props.theme.typography.fontFamily;
});
/**
 * Content Wrapper.
 */

var Content = styled.div(_templateObject2$5(), function (props) {
  return props.theme.typography.fontFamily;
});
/**
 * Title of the Card.
 */

var Header$2 = styled.div(_templateObject3$2(), function (props) {
  return props.theme.typography.fontFamily;
});
/**
 * Card meta block.
 */

var Meta = styled.div(_templateObject4$1(), function (props) {
  return props.theme.typography.fontFamily;
});
/**
 * Cointains the major content of a Card.
 */

var Description = styled.div(_templateObject5(), function (props) {
  return props.theme.typography.fontFamily;
}); // Props

Card.propTypes = {
  rounded: propTypes.PropTypes.bool
};
Card.defaultProps = {
  theme: DefaultTheme // Assign children

};
Card.Content = Content;
Card.Header = Header$2;
Card.Meta = Meta;
Card.Description = Description;

function _templateObject4$2() {
  var data = _taggedTemplateLiteral(["\n  color: rgba(0, 0, 0, 0.87);\n  font-family: ", ";\n  font-size: 1em;\n  line-height: 1.4285em;\n  margin-top: 0.6em;\n"]);

  _templateObject4$2 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3$3() {
  var data = _taggedTemplateLiteral(["\n  color: rgba(0, 0, 0, 0.85);\n  display: inline-block;\n  font-family: ", ";\n  font-weight: bold;\n  margin: -0.21425em 0em 0em;\n"]);

  _templateObject3$3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2$6() {
  var data = _taggedTemplateLiteral(["\n  background: none;\n  border-radius: 0em;\n  border: none;\n  box-shadow: none;\n  display: block;\n  font-family: ", ";\n  flex: 1 1 auto;\n  font-size: 1em;\n  margin: 0em;\n  padding: 0em;\n  -ms-flex: 1 1 auto;\n  -webkit-box-flex: 1;\n"]);

  _templateObject2$6 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$e() {
  var data = _taggedTemplateLiteral(["\n  background: transparent;\n  border-radius: 0rem;\n  border: none;\n  box-shadow: none;\n  display: -ms-flexbox;\n  display: -webkit-box;\n  display: flex;\n  font-family: ", ";\n  margin: 1em 0em;\n  min-height: 0px;\n  padding: 0em;\n  transition: box-shadow 0.1s ease;\n  width: 100%;\n  -webkit-transition: box-shadow 0.1s ease;\n"]);

  _templateObject$e = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Item component.
 */

var Item = styled.div(_templateObject$e(), function (props) {
  return props.theme.typography.fontFamily;
});
var Content$1 = styled.div(_templateObject2$6(), function (props) {
  return props.theme.typography.fontFamily;
});
var Header$3 = styled.div(_templateObject3$3(), function (props) {
  return props.theme.typography.fontFamily;
});
var Text$1 = styled.div(_templateObject4$2(), function (props) {
  return props.theme.typography.fontFamily;
}); // Default Theme

Item.defaultProps = {
  theme: DefaultTheme
};
Content$1.defaultProps = {
  theme: DefaultTheme
};
Header$3.defaultProps = {
  theme: DefaultTheme
};
Text$1.defaultProps = {
  theme: DefaultTheme // Assign children

};
Item.Content = Content$1;
Item.Header = Header$3;
Item.Text = Text$1;

function _templateObject$f() {
  var data = _taggedTemplateLiteral(["\n  background-color: ", ";\n  //border: 2px solid\n  //   ", ";\n  border: none;\n  border-radius: ", ";\n  cursor: pointer;\n  display: inline-block;\n   color: ", ";\n  font-family: ", ";\n  font-size: 1em;\n  font-style: normal;\n  font-weight: bold;\n  line-height: 1em;\n  margin: 1.5px;\n  min-height: 1em;\n  padding: 0.6875em 1.5em;\n  text-align: center;\n  text-decoration: none;\n  vertical-align: baseline;\n  &:hover {\n     background-color: ", ";\n  }\n"]);

  _templateObject$f = function _templateObject() {
    return data;
  };

  return data;
}

var chroma$1 = require('chroma-js');
/**
 * Button component. This renders a html `button`, not an `input` with type `submit` as `button` is more flexible.
 * HTML: `button`
 * @type {StyledComponentClass<JSX.IntrinsicElements["button"], any, JSX.IntrinsicElements["button"]>}
 */


var Button = styled.button(_templateObject$f(), function (props) {
  return props.color ? props.theme.color[props.color] : props.theme.color.grey;
}, function (props) {
  return props.color ? props.color : props.theme.primaryColor;
}, function (_ref) {
  var rounded = _ref.rounded;
  return rounded ? '.15rem' : 0;
}, function (props) {
  return chroma$1.contrast(props.theme.color[props.color] || 'white', props.theme.typography.fontColor) > 7 ? props.theme.typography.fontColor : 'white';
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return chroma$1(props.theme.color[props.color] || props.theme.color.grey).darken();
}); // Props

Button.propTypes = {
  rounded: propTypes.PropTypes.bool // Default Theme

};
Button.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$g() {
  var data = _taggedTemplateLiteral(["\n  font-size: ", ";\n  font-family: ", ";\n"]);

  _templateObject$g = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Renders a `code` block
 */

var Code = styled.code(_templateObject$g(), function (props) {
  return props.fontSize ? props.fontSize : props.theme.typography.fontSize;
}, function (props) {
  return props.theme.typography.fontFamily;
});
Code.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$h() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n  display: block;\n  font-family: ", ";\n  font-size: ", ";\n  font-weight: ", ";\n"]);

  _templateObject$h = function _templateObject() {
    return data;
  };

  return data;
}

var setFontSize = function setFontSize(value, size) {
  var sizes = {
    h1: "calc(".concat(size, "*1)"),
    h2: "calc(".concat(size, " * 0.9)"),
    h3: "calc(".concat(size, " * 0.8)"),
    h4: "calc(".concat(size, " * 0.7)"),
    h5: "calc(".concat(size, " * 0.6)"),
    h6: "calc(".concat(size, " * 0.5)"),
    default: "calc(".concat(size, "*1)")
  };
  return sizes[value] || sizes.default;
};

var Heading = styled.h1(_templateObject$h(), function (props) {
  return props.color ? props.color : props.theme.typography.fontColor;
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return setFontSize(props.as, props.theme.heading.fontSize);
}, function (props) {
  return props.theme.heading.fontWeight;
});

var Element = function Element(props) {
  return React.createElement(Heading, _extends({
    as: "h".concat(props.level || 1)
  }, props), props.children);
};

Heading.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$i() {
  var data = _taggedTemplateLiteral(["\n  display: block;\n  flex: 0 0 auto;\n  height: auto;\n  padding: 0;\n  position: relative;\n  max-width: 100%;\n  -webkit-box-flex: 0;\n"]);

  _templateObject$i = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Image component.
 * HTML: `img`
 */

var Image = styled.img(_templateObject$i());

function _templateObject$j() {
  var data = _taggedTemplateLiteral(["\n  border: 1px inset ", ";\n  border-radius: ", ";\n  display: -webkit-inline-box;\n  display: -ms-inline-flexbox;\n  display: inline-flex;\n  font-family: ", ";\n  font-size: 1rem;\n  line-height: 1.2rem;\n  margin-right: 0.1rem;\n  outline: 0;\n  padding: 0.4rem 0.5rem;\n  &:focus {\n    background: #ffffff;\n    box-shadow: none;\n    outline: 1px solid ", ";\n    -webkit-box-shadow: none;\n  }\n"]);

  _templateObject$j = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Input field, defaults to `text`.
 * HTML: `input`
 * @type {StyledComponentClass<*, any, *>}
 */

var Input = styled.input.attrs({
  type: 'text'
})(_templateObject$j(), function (props) {
  return props.theme.color.grey;
}, function (props) {
  return props.rounded ? '0.125rem' : null;
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return props.theme.color.secondary;
});
Input.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$k() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n  font-family: ", ";\n  font-size: 1rem;\n  text-decoration: ", ";\n"]);

  _templateObject$k = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Link element
 * HTML: `a`
 * @type {StyledComponentClass<*, any, *>}
 */

var Link = styled.a.attrs({
  href: function href(props) {
    return props.to || null;
  },
  rel: function rel(props) {
    return props.external ? 'noopener noreferrer' : null;
  },
  target: function target(props) {
    return props.external ? '_blank' : '_self';
  }
})(_templateObject$k(), function (props) {
  return props.color || props.theme.typography.fontColor;
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return props.underlined ? 'underline' : 'none';
});
Link.defaultProps = {
  theme: DefaultTheme
};

function _templateObject2$7() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n"]);

  _templateObject2$7 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$l() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n"]);

  _templateObject$l = function _templateObject() {
    return data;
  };

  return data;
}
var List = styled.ul(_templateObject$l(), function (props) {
  return props.theme.typography.fontFamily;
});
var ListItem = styled.li(_templateObject2$7(), function (props) {
  return props.theme.typography.fontFamily;
});

var Element$1 = function Element(props) {
  return React.createElement(List, _extends({
    as: "".concat(props.ordered ? 'ol' : 'ul')
  }, props));
};

Element$1.defaultProps = {
  theme: DefaultTheme
};
ListItem.defaultProps = {
  theme: DefaultTheme
};
Element$1.Item = ListItem;

function _templateObject$m() {
  var data = _taggedTemplateLiteral(["\n  font-size: ", ";\n  color: ", ";\n"]);

  _templateObject$m = function _templateObject() {
    return data;
  };

  return data;
}
var Span = styled.span(_templateObject$m(), function (props) {
  return props.fontSize || props.theme.fontSize;
}, function (props) {
  return props.color || props.theme.typography.fontColor;
});
Span.defaultProps = {
  theme: DefaultTheme
};

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n  border: 1px solid ", ";\n  font-family: ", ";\n  padding: 0.5rem;\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  border: 1px solid ", ";\n  font-family: ", ";\n  padding: 0.5rem;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5$1() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n"]);

  _templateObject5$1 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4$3() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  color: ", ";\n"]);

  _templateObject4$3 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3$4() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  color: ", ";\n"]);

  _templateObject3$4 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2$8() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  color: ", ";\n"]);

  _templateObject2$8 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject$n() {
  var data = _taggedTemplateLiteral(["\n  border: 1px solid ", ";\n  border-collapse: collapse;\n  color: ", ";\n  font-family: ", ";\n"]);

  _templateObject$n = function _templateObject() {
    return data;
  };

  return data;
}
var Table = styled.table(_templateObject$n(), function (props) {
  return props.theme.color.grey;
}, function (props) {
  return props.color || props.theme.typography.fontColor;
}, function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
});
var TableHeader = styled.thead(_templateObject2$8(), function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
}, function (props) {
  return props.color || props.theme.typography.fontColor;
});
var TableBody = styled.tbody(_templateObject3$4(), function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
}, function (props) {
  return props.color || props.theme.typography.fontColor;
});
var TableFooter = styled.tfoot(_templateObject4$3(), function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
}, function (props) {
  return props.color || props.theme.typography.fontColor;
});
var TableRow = styled.tr(_templateObject5$1(), function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
});
var TableHeading = styled.th(_templateObject6(), function (props) {
  return props.theme.color.grey;
}, function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
});
var TableCell = styled.td(_templateObject7(), function (props) {
  return props.theme.color.grey;
}, function (props) {
  return props.fontFamily || props.theme.typography.fontFamily;
});
Table.defaultProps = {
  theme: DefaultTheme
};
TableHeader.defaultProps = {
  theme: DefaultTheme
};
TableBody.defaultProps = {
  theme: DefaultTheme
};
TableFooter.defaultProps = {
  theme: DefaultTheme
};
TableRow.defaultProps = {
  theme: DefaultTheme
};
TableHeading.defaultProps = {
  theme: DefaultTheme
};
TableCell.defaultProps = {
  theme: DefaultTheme
};
Table.Header = TableHeader;
Table.Body = TableBody;
Table.Footer = TableFooter;
Table.Row = TableRow;
Table.Heading = TableHeading;
Table.Cell = TableCell;

function _templateObject$o() {
  var data = _taggedTemplateLiteral(["\n  color: ", ";\n  font-family: ", ";\n  font-size: ", ";\n  line-height: 1.5rem;\n"]);

  _templateObject$o = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * Text element
 * HTML: `p`
 * @type {StyledComponentClass<JSX.IntrinsicElements["p"], any, JSX.IntrinsicElements["p"]>}
 */

var Text$2 = styled.p(_templateObject$o(), function (props) {
  return props.color || props.theme.typography.fontColor;
}, function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return props.theme.typography.fontSize;
});
Text$2.defaultProps = {
  theme: DefaultTheme
};

function _templateObject$p() {
  var data = _taggedTemplateLiteral(["\n  font-size: 1rem;\n  font-family: ", ";\n  text-align: ", ";\n"]);

  _templateObject$p = function _templateObject() {
    return data;
  };

  return data;
}
/**
 * View element
 * HTML: `div`
 * @type {StyledComponentClass<JSX.IntrinsicElements["p"], any, JSX.IntrinsicElements["p"]>}
 */

var View = styled.div(_templateObject$p(), function (props) {
  return props.theme.typography.fontFamily;
}, function (props) {
  return props.centered ? 'center' : 'left';
});
View.defaultProps = {
  theme: DefaultTheme
};

// TODO use named imports

/* Themes */

exports.DefaultTheme = DefaultTheme;
exports.StyledComponentsTheme = Theme;
exports.Grid = Grid;
exports.Menu = Menu;
exports.Message = Message;
exports.Badge = Badge;
exports.Container = Container;
exports.Dropdown = Dropdown;
exports.Header = Header$1;
exports.Label = Label;
exports.Loader = Loader;
exports.NavBar = NavBar;
exports.Pagination = Pagination;
exports.Segment = Segment;
exports.Sidebar = Sidebar;
exports.Card = Card;
exports.Item = Item;
exports.Button = Button;
exports.Code = Code;
exports.Heading = Element;
exports.Image = Image;
exports.Input = Input;
exports.Link = Link;
exports.List = Element$1;
exports.Span = Span;
exports.Table = Table;
exports.Text = Text$2;
exports.View = View;
